lib/buildpackutil.py

```diff
@@ -265,38 +275,29 @@ def _checkout_from_git_rootfs(directory, mx_version):
     )
 
 
-def fix_mono_config_and_get_env(dot_local_location, mono_lib_dir):
+def _get_env_with_monolib(mono_lib_dir):
     env = dict(os.environ)
-    env['LD_LIBRARY_PATH'] = mono_lib_dir
-
-    if not os.path.isfile(os.path.join(mono_lib_dir, 'libgdiplus.so')):
+    env['LD_LIBRARY_PATH'] = mono_lib_dir + '/lib'
+    if not os.path.isfile(os.path.join(mono_lib_dir + '/lib/', 'libgdiplus.so')):
         raise Exception('libgdiplus.so not found in dir %s' % mono_lib_dir)
-
-    subprocess.check_call([
-        'sed',
-        '-i',
-        's|/app/vendor/mono/lib/libgdiplus.so|%s|g' % os.path.join(
-            mono_lib_dir, 'libgdiplus.so'
-        ),
-        os.path.join(_get_mono_path(dot_local_location), 'etc/mono/config'),
-    ])
-    subprocess.check_call([
-        'sed',
-        '-i',
-        's|/usr/lib/libMonoPosixHelper.so|%s|g' % os.path.join(
-            mono_lib_dir, 'libMonoPosixHelper.so'
-        ),
-        os.path.join(_get_mono_path(dot_local_location), 'etc/mono/config'),
-    ])
     return env
 
 
-def _get_mono_path(dot_local_location):
+def _detect_mono_version():
+    if os.environ.get('FORCED_MONO4_VERSION'):
+        logging.warning('Using forced mono version')
+        target = 'mono-4.4.1.0'
+    else:
+        target = 'mono-3.10.0'
+    logging.info('Selecting Mono Runtime: ' + target)
+    return target
+
+
+def _get_mono_path(directory, mono_version):
     return get_existing_directory_or_raise([
-        os.path.join(dot_local_location, 'mono'),
-        '/opt/mono-3.10.0',
-        '/usr/local/share/mono-3.10.0',
```

Comment:
AFAIK this line should stay here for backwards compatibility with previous rootfs'es, since we specifically create a symlink while building the rootfs, see commit `0e8e5db423c9e57a4c641646fc51b6a47a701340` in Gitlab `cf-tools-bash`, file `mxlinuxfs2/mxrootfs.sh`.